### ISSUE TITLE
Provide links to the specification documents that Mastodon uses

### ISSUE THREAD

**Issue by hach-que** [Gargron/mastodon#188]
I've been trying to track down specification documents for the protocols Mastodon uses, and in some cases I'm coming up empty handed.  So far what I've managed to nail down is:

* **OStatus**
  * _Spec:_ https://www.w3.org/community/ostatus/wiki/images/9/93/OStatus_1.0_Draft_2.pdf
  * _XSD or XML Specification:_ The schema URL used as the XML namespace seems to 404???
* **Portable Contacts**
  * _Spec:_ https://web.archive.org/web/20160305010620/http://portablecontacts.net/draft-spec.html (seems to be only available on Web Archive???)
* **Atom**
  * _Spec_: https://www.w3.org/2005/Atom
* **Atom Activity Streams**
  * _Spec_: http://activitystrea.ms/specs/atom/1.0/

I haven't looked around for others yet, I'm trying to determine whether there's a more detailed version of OStatus first.

Basically I was looking at implementing lists using groups, which OStatus does define (section 3 Definitions), but then provides no guidance as the structure or content of what a "group" should be or have, whether that's in terms of data or it's representation in XML.  So I'm quite lost on exactly how to proceed with representing that structure in a way that works in a federated environment.

**Comment by Gargron** [comment]
I decided not to implement groups in Mastodon because I personally didn't see much use in it over hashtags (also dislike the !bang syntax). But technically a group account is just like a user account, also with an Atom feed, only difference being all the `<entry>`s have different `<author>`s because they come from all users in the group and not from the group account. And then I'm guessing a group Salmon would forward the message to all group members, but I'm not sure, just a guess

I'm actually really grateful that you found some of the links to the RFC, it's not an easy feat. I somehow managed to find them when I did the first prototype ~6 months ago, but a lot of the time I was literally looking at output from GNU social to match it.

**Comment by hach-que** [comment]
I think it would probably be good if we mirrored some of these specs in the repository.  At least one of them (Portable Contacts) seems to have completely disappeared off the internet - even the home page doesn't work any more.  If we keep copies around, it at least insulates us somewhat against other specs disappearing.

That said - I'm personally a little bit sceptical of maintaining compliance with specifications which are clearly not being maintained any more and have no working group to head them.  I think in those cases Mastodon in the long term should either try to take on maintenance and revisions of those protocols, or design and implement alternative protocols that provide a super set of the functionality of the original spec.

**Comment by Gargron** [comment]
I agree with the sentiment and I certainly felt disappointment in the maintenance of those documents especially considering GNU social is still alive. However at least in the case of Portable Contacts there's not much to look at, it's just `<poco:displayName>` for the display name, `<poco:note>` for bio and a few others.

**Comment by nex3** [comment]
I think https://github.com/OStatus/spec might be the current canonical location of the OStatus spec?

**Comment by Gargron** [comment]
Okay, just added a wiki page https://github.com/Gargron/mastodon/wiki/Specs-and-RFCs-used

**Comment by jjatria** [comment]
Since the wiki has disappeared, this is just to update the link in the previous comment:

https://github.com/tootsuite/documentation/blob/master/Specs-and-RFCs-used.md
